File: CodeChef/DecCookOff/PRFYIT.py
import sys


def find_min_count(s):

    start = 0
    end = start + 1

    z_one_cnt = 0
    o_zero_cnt = 0

    # 0101
    while end < len(s):
        while start < len(s) and s[start] == '1':
            start += 1

        end = start + 1

        while end < len(s) and s[end] == '0':
            end += 1

        if end < len(s):
            z_one_cnt += 1

        start = end

    # 1010
    start = 0
    while end < len(s):
        while start < len(s) and s[start] == '0':
            start += 1

        end = start + 1

        while end < len(s) and s[end] == '1':
            end += 1

        if end < len(s):
            o_zero_cnt += 1

        start = end

    return max(0, max(z_one_cnt, o_zero_cnt) - 1)


def main():

    t = int(input())

    while t > 0:
        t -= 1
        s = sys.stdin.readline().strip()

        print(find_min_count(s))


def custom_test():
    s = "0110"
    print(find_min_count(s))


if __name__ == '__main__':
    main()
    # custom_test()

File: CodeChef/DecCookOff/PRFYIT.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <string>

using namespace std;

int countMinDelete(string s) {
    int zOneCnt = 0;
    int oZeroCnt = 0;

    int start = 0;
    int end;

    // 0101
    while (end < s.length()) {

        while (start < s.length() && s[start] == '1') {
            start++;
        }

        end = start + 1;

        while (end < s.length() && s[end] == '0') {
            end++;
        }

        if (end < s.length()) {
            zOneCnt++;
        }
        start = end;
    }

    start = 0;
    // 1010
    while (end < s.length()) {

        while (start < s.length() && s[start] == '0')
        {
            start++;
        }
        end = start + 1;
        
        while (end < s.length() && s[end] == '1') {
            end++;
        }

        if (end < s.length()) {
            oZeroCnt++;
        }
        start = end;
    }

    return max(oZeroCnt, zOneCnt) - 1;
}

int main() {

    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        cout << countMinDelete(s) << endl;
    }

    return 0;
}